### PR TITLE
#380 harden the reset triggers

### PR DESCRIPTION
**Pull request by krazykaze81**
Keep the stats in sync with all players - do not use sess.rounds for stopwatch

### CODE REVIEW

File: src/game/g_session.c [hunk 1]
```diff
@@ -409,19 +409,21 @@ void G_WriteSessionData( void ) {
 
 	// L0 - OSP Stats
 	// Keep stats for all players in sync
-	for ( i = 0; !level.fResetStats && i < level.numConnectedClients; i++ ) {
-		if ( ( g_gamestate.integer == GS_WARMUP_COUNTDOWN &&
-			   ( ( g_gametype.integer == GT_WOLF_STOPWATCH && level.clients[level.sortedClients[i]].sess.rounds >= 2 ) ||
-				 ( g_gametype.integer != GT_WOLF_STOPWATCH && level.clients[level.sortedClients[i]].sess.rounds >= 1 ) ) ) ) {
+	if (!level.fResetStats && level.numConnectedClients > 0) {
```
Comment (rtcw-nihi) [inline src/game/g_session.c:412]: should be looping over all clients as it was

File: src/game/g_session.c [hunk 2]
```diff
@@ -409,19 +409,21 @@ void G_WriteSessionData( void ) {
...
+		if ((g_gamestate.integer == GS_WARMUP_COUNTDOWN &&
+			((g_gametype.integer == GT_WOLF_STOPWATCH && g_currentRound.integer == 0) ||
+				(g_gametype.integer != GT_WOLF_STOPWATCH && level.clients[level.sortedClients[0]].sess.rounds >= 1)))) {
```
Comment (rtcw-nihi) [inline src/game/g_session.c:415]: should be looping over all clients as it was

**Comment by donkz** [comment]
What if some people have 1 round and some 2 - that means stats will reset 100% of the time, which i believe more and more is the exact issue

g_gametype.integer == GT_WOLF_STOPWATCH && level.clients[level.sortedClients[i]].sess.rounds >= 2

**Comment by rtcw-nihi** [comment]
keep the addition `((g_gametype.integer == GT_WOLF_STOPWATCH && g_currentRound.integer == 0) ` but for non-stopwatch we keep the original 
` ( g_gametype.integer != GT_WOLF_STOPWATCH && level.clients[level.sortedClients[i]].sess.rounds >= 1 ) ) )` and the for loop

**Comment by rtcw-nihi** [comment]
i.e.
	for ( i = 0; !level.fResetStats && i < level.numConnectedClients; i++ ) {
		if ( ( g_gamestate.integer == GS_WARMUP_COUNTDOWN &&
			   ( (g_gametype.integer == GT_WOLF_STOPWATCH && g_currentRound.integer == 0) ||
				 ( g_gametype.integer != GT_WOLF_STOPWATCH && level.clients[level.sortedClients[i]].sess.rounds >= 1 ) ) ) ) {
			level.fResetStats = qtrue;
		}
	} // End

**Comment by krazykaze81** [comment]
Made some changes for your review @rtcw-nihi

**Pull request merged**